my-webxr-app/src/data/DataLayer.tsx

```diff
@@ -0,0 +1,330 @@
+      });
+
+      await Promise.all(promises);
+      return true;
+    } catch (error) {
+      return Promise.resolve(false);
+    }
+  }
+
+  // TODO: query column names from stats table for PCA and let user choose subset to do pca on
+
+  /**
+   * Retrieves column data for PCA. (Raw data or standardized data columns)
+   *
+   * This function accepts an array of column names, retrieves the corresponding data for each
+   * column based on the specified column type, and returns a new Matrix instance with the retrieved
+   * data.
+   *
+   * The Matrix size could be too large and exceed the memory limit. TODO handle this case.
+   *
+   * If an error occurs during the operation (e.g., a column does not exist, table empty), the
+   * function will catch the error and return an empty Matrix.
+   *
+   * @param {string[]} columnNames - The names of the columns to retrieve data for.
+   * @param {ColumnType} columnType - The type of the columns to retrieve data for.
+   * @returns {Promise<Matrix>} A promise that resolves to a Matrix instance containing the
+   * retrieved data.
+   *
+   * @throws {Error} If the columnType is not RAW or STANDARDIZED.
+   */
+  async getColumnsForPca(columnNames: string[], columnType: ColumnType): Promise<Matrix> {
+    if (columnType !== ColumnType.RAW && columnType !== ColumnType.STANDARDIZED) {
+      throw new Error('Invalid column type. Must be either RAW or STANDARDIZED.');
+    }
+
+    try {
+      const columnDataArray: number[][] = [];
+
+      // Assume all the columns are numeric
+      const promises = columnNames.map((columnName) => this.repository
+        .getColumn(columnName, columnType)
+        .then((columnData) => columnDataArray.push(columnData.values as number[])));
+
+      await Promise.all(promises);
+
+      return new Matrix(columnDataArray);
+    } catch (error) {
+      // Logging the error here
+      return new Matrix([]);
+    }
+  }
+
+  /**
+   * TODO - modify PCA column to store varience explained or eignevalues
+   * Write back the PCA column to the repository. Name column in terms of their significant.
+   *
+   * @returns {Promise<boolean>} A promise that resolves to `true` if the operation was successful,
+   */
+  // temp disable
+  // eslint-disable-next-line class-methods-use-this
+  async storePCA(): Promise<boolean> {
+    return Promise.resolve(true);
+  }
+
+  /**
+   * TODO
+   * Select a raw data column or PCA column from the repository for graphing
+   */
+  // temp disable
+  // eslint-disable-next-line class-methods-use-this
+  async selectAxes() {
+    return Promise.resolve(true);
+  }
+
+  /**
+   * TODO
+   * Select a PCA column from the repository
+   */
+  // temp disable
+  // eslint-disable-next-line class-methods-use-this
+  async selectPCA() {
+    return Promise.resolve(true);
+  }
+}
+
+/**
+ * The BatchedDataStream type is used for streaming in batches of data from CSV parsing.
+ */
+export type BatchedDataStream = Array<DataRow>;
```

Comment:
minor: I think BatchedDataStream makes more sense to be in CsvUtils as batches are generated in CsvUtils. That way we can avoid dependency cycle

Reply:
Yeah, similar to what I alluded to above. We may want to break out these into into their own file in the `types` directory.